packages/grafana-ui/src/components/DataTable/types.ts

```diff
@@ -0,0 +1,34 @@
+import { ReactNode } from 'react';
+import { CellProps, DefaultSortTypes, IdType, SortByFn } from 'react-table';
+
+export interface Column<TableData extends object> {
+  /**
+   * ID of the column.
+   * Set this to the matching object key of your data or `undefined` if the column doesn't have any associated data with it.
+   * This must be unique among all other columns.
+   */
+  id?: IdType<TableData>;
+  /**
+   * Custom render function for te cell
+   */
+  cell?: (props: CellProps<TableData>) => ReactNode;
+  /**
+   * Header name. if `undefined` the header will be empty. Useful for action columns.
+   */
+  header?: string;
+  /**
+   * Column sort type. If `undefined` the column will not be sortable.
+   * */
+  sortType?: DefaultSortTypes | SortByFn<TableData>;
+  /**
+   * If `true` the column will shrink to its minimum width.
+   */
+  shrink?: boolean;
```

Comment:
This naming here isn't very clear to me, I thought it was related to flex-shrink. The behavior is actually more of a `grow: false`, wdyt about calling it `canGrow` and reverse the `boolean` meaning?

Reply:
i personally think optional boolean props/options are alway easier to reason about when their default is `false` and by setting them to `true` you trigger a behaviour that wouldn't otherwise be triggered.

in this case, being it an optional boolean property i would expect not passing it to behave the same as setting it to false.

while i agree that `shrink` may not be the best name, i always have an hard time having to "inverse" how i think about props/options when some of them reverse this principle.

what about `disableGrow`? or maybe `narrow` (which i kinda like as it sounds more declarative)?

Reply:
> i personally think optional boolean props/options are alway easier to reason about when their default is false and by setting them to true you trigger a behaviour that wouldn't otherwise be triggered.
Yep you're right those do make more sense!

I personally prefer `disableGrow` to `narrow`, because with `narrow` it makes it sound like the column will be really tiny, when really it can be gigantic depending on the title 😂  But maybe it's just a personal preference.

Reply:
good point. i agree 👍